containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/org/hawkular/openshift/namespace/NamespaceOverrideMapper.java

```diff
@@ -0,0 +1,271 @@
+            }
+        }
+
+        Cluster cluster = clusterBuilder.build();
+        cluster.init();
+
+        Session session = cluster.connect();
+
+        return session;
+    }
+
+    /**
+     * Creates a keyspace if it does not already exist
+     * @param keyspace The name of the keyspace
+     */
+    private void ensureKeyspace(String keyspace) {
+        ResultSet resultset = session.execute("SELECT * FROM system_schema.keyspaces WHERE keyspace_name = '" + keyspace + "';");
+        if (resultset.all().isEmpty()) {
+            session.execute("CREATE KEYSPACE " + keyspace + " WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'}  AND durable_writes = true;");
+        }
+    }
+
+    private boolean hasMappingTable() {
+        ResultSet resultset = session.execute("SELECT table_name FROM system_schema.tables WHERE keyspace_name='openshift_metrics' AND table_name='metrics_mappings';");
+        return !resultset.all().isEmpty();
+    }
+
+    private void createMappingTable() {
+        ResultSet resultset = session.execute("SELECT table_name FROM system_schema.tables WHERE keyspace_name='openshift_metrics' AND table_name='metrics_mappings';");
+        if (resultset.all().isEmpty()) {
+            session.execute("CREATE TABLE openshift_metrics.metrics_mappings (project_id text PRIMARY KEY, project_name text);");
+        }
+    }
+
+    private Map<String,String> getMappingTable() {
+        Map<String,String> mappings = new HashMap<>();
+        if (hasMappingTable()) {
+            ResultSet resultSet = session.execute("SELECT project_id,project_name FROM openshift_metrics.metrics_mappings;");
+
+            for (Row row: resultSet.all()) {
+                mappings.put(row.getString("project_id"), row.getString("project_name"));
+            }
+
+        }
+
+        return mappings;
+    }
+
+    private void writeMappingTable(Map<String,String> mappings) {
+        createMappingTable();
+
+        StringBuffer buffer = new StringBuffer();
+        buffer.append("BEGIN BATCH ");
+
+        for (Map.Entry<String,String> entry: mappings.entrySet()) {
+            buffer.append(" INSERT INTO openshift_metrics.metrics_mappings (project_id, project_name) VALUES('"+ entry.getKey() + "','" + entry.getValue() + "');");
+        }
+        buffer.append(" APPLY BATCH;");
+
+        ResultSet resultset = session.execute(buffer.toString());
+
+        System.err.println("WRITE RESULT SET");
+    }
+
+    private Set<String> getTenants() {
+        Set<String> tenants = new HashSet<String>();
+        ResultSet resultset = session.execute("SELECT * FROM system_schema.keyspaces WHERE keyspace_name = '" + "hawkular_metrics" + "';");
+        if (resultset.all().isEmpty()) {
+            return tenants;
+        }
+
+        ResultSet resultSet = session.execute("SELECT DISTINCT tenant_id,type from hawkular_metrics.metrics_idx;");
+
+        for (Row row:resultSet.all()) {
```

Comment:
``resultSet.all()`` instead of iterator could return a very large batch of data. Not recommended.